### PR TITLE
Add several new options to Rake task

### PR DESCRIPTION
**Pull request by shortdudey123**
Adds and documents 3 new options for the rake task:
- debug
- exclude_paths
- fail_on_error

### CODE REVIEW

**Comment by shortdudey123** [comment]
@dougbarth can you take a look at this when you have a chance? Thanks!

File: lib/jsonlint/rake_task.rb [hunk 1]
```diff
@@ -24,7 +30,14 @@ def define_task
       task(name) do
         puts 'Running JsonLint...'
 
-        files_to_check = Rake::FileList.new(paths)
+        YamlLint.logger.level = Logger::DEBUG if debug
+
+        files_to_check_raw = Rake::FileList.new(paths)
+        files_to_exclude = Rake::FileList.new(exclude_paths)
+        files_to_check = files_to_check_raw - files_to_exclude
```
Comment (dougbarth) [inline lib/jsonlint/rake_task.rb]: Any reason not to use `Rake::FileList#exclude` to incrementally build up the `FileList` to iterate over? I'm a bit worried that this approach will have subtle behavioral or performance differences.

File: lib/jsonlint/rake_task.rb [hunk 2]
```diff
@@ -24,7 +30,14 @@ def define_task
       task(name) do
         puts 'Running JsonLint...'
 
-        files_to_check = Rake::FileList.new(paths)
+        YamlLint.logger.level = Logger::DEBUG if debug
```
Comment (dougbarth) [inline lib/jsonlint/rake_task.rb]: Instead of exposing a boolean for debug, I would suggest exposing the log level directly.

**Comment by shortdudey123** [comment]
Fixed both comments :)

**Comment by dougbarth** [comment]
LGTM

**Comment by shortdudey123** [comment]
Thanks!

**Pull request merged**